lib/spack/spack/schema/resource.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

Comment:
update to 2024

Can we call this something else to distinguish it from the `resource` directive?

`binary_resource`/`bootstrap_resource`. I can't think of a name that isn't two part. FWIW I don't think there's a need to convert from `-` to `_` (e.g. we can force users to `spack config blame bootstrap_resource` vs. `... bootstrap-resource`).